### PR TITLE
test(integration): add access logging test

### PR DESCRIPTION
**Pull request by DanStough**
### Description
Adding an integration test for access logs. It does not comprehensively execute the API, but it does demonstrate a possible workflow.

### Testing & Reproduction steps
* Use go test to run the integration test.

### PR Checklist

* [X] updated test coverage
* [ ] ~external facing docs updated~
* [X] not a security concern

### CODE REVIEW

File: test/integration/consul-container/libs/assert/service.go
Comment (DanStough) [inline test/integration/consul-container/libs/assert/service.go:34]: Reordering alphabetically to make this easier to navigate

File: test/integration/consul-container/libs/assert/service.go
Comment (DanStough) [inline test/integration/consul-container/libs/assert/service.go:15]: My editor sorts these according to the convention I've heard we use.

File: test/integration/consul-container/libs/cluster/container.go
Comment (DanStough) [inline test/integration/consul-container/libs/cluster/container.go:68]: I started going this route to bang on the public listener and generate access logs. Since I ended up not including this in the test because there is more to Envoy's internal responses than I thought, we could take or leave this.

File: test/integration/consul-container/libs/service/helpers.go
Comment (thisisnotashwin) [inline test/integration/consul-container/libs/service/helpers.go:19]: ❤️ love this so much!

File: test/integration/consul-container/test/observability/access_logs_test.go
Comment (thisisnotashwin) [inline test/integration/consul-container/test/observability/access_logs_test.go:36]: TYSM for the steps!

File: test/integration/consul-container/test/observability/access_logs_test.go
Comment (thisisnotashwin) [inline test/integration/consul-container/test/observability/access_logs_test.go:56]: lol

**Review by thisisnotashwin — APPROVED** [review]
This is both very clearly written and a fun read where it could get dreary!! LGTM if the test goes 🟢

File: test/integration/consul-container/test/observability/access_logs_test.go
Comment (huikang) [inline test/integration/consul-container/test/observability/access_logs_test.go:119]: I think we can replace this method with a call to https://github.com/hashicorp/consul/blob/a1e2a4f8d6bef92aaddcabd42d046b258dbffb30/test/integration/consul-container/libs/topology/peering_topology.go#L152

it works for me in another [PR](https://github.com/hashicorp/consul-enterprise/pull/4091/files#diff-49f24a7c95c856d2b708203061008c49c3962dbe2f577d945508f72335465452R24)

**Review by huikang — APPROVED** [review]
One minor comment regarding reusing a cluster creation method.

File: test/integration/consul-container/libs/cluster/container.go
Comment (DanStough) [reply test/integration/consul-container/libs/cluster/container.go]: I decided to remove this. It's just more noise and it makes the re-ordering I did between starting the sidecar and service registration more challenging.

File: test/integration/consul-container/test/observability/access_logs_test.go
Comment (DanStough) [reply test/integration/consul-container/test/observability/access_logs_test.go]: Nice! In the future I think we should remove peering from the name since it's good for everything.

**Pull request merged**
